src/kv/Makefile.am

```diff
@@ -16,12 +16,17 @@ if WITH_SLIBROCKSDB
 # build rocksdb with its own makefile
 # for some stupid reason this needs -fPIC...
 # PORTABLE=1 fixes the aarch64 build (-march=native doesn't work there)
-NPROC = nproc
+
+ROCKSDBCXX_FLAGS = "-fPIC"
 if FREEBSD
-        NPROC = sysctl -n hw.ncpu
+NPROC = sysctl -n hw.ncpu
+ROCKSDBCXX_FLAGS += "-Wmismatched-tags"
+else
+NPROC = nproc
 endif
+
 rocksdb/librocksdb.a:
-	cd rocksdb && CC="${CC}" CXX="${CXX}" EXTRA_CXXFLAGS=-fPIC PORTABLE=1 $(MAKE) -j$(shell ${NPROC}) static_lib
+	cd rocksdb && CC="${CC}" CXX="${CXX}" EXTRA_CXXFLAGS="${ROCKSDBCXX_FLAGS}" PORTABLE=1 ${MAKE} -j$(shell ${NPROC}) static_lib
```

Comment:
@trociny 
I don't really get the point you are making?
For Clang/FreeBSD we add -Wmismatched-tags (which has an error, as I now see)
And I'm using your suggestion to fetch NPROC from sysctl